### PR TITLE
Merge data model pages

### PR DESCRIPTION
**Pull request by snide**
This moves the `/postgres/data-model` page to be included in the top level data-model page to better highlight what the Xata model does and how it interacts with postgres.

This also removes the object type, which was deprecated a long time ago.

### CODE REVIEW

File: 030-Data-types/030-data-model.mdx [hunk 1]
```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
 published: true
 ---
 
+Xata builds on top of Postgres and thus supports the [default data types](https://www.postgresql.org/docs/current/datatype.html) inherit in that platform. Xata additionally adds to that list with unique types such as `file`, `filearray` and `vector` to handle platform features like [file attachements](/docs/sdk/file-attachments) and [vector searching](/docs/sdk/vector-search). In addition, there are some added types to simplify validation of common `text` patterns such as `email` and `multiple`. Usage of these types is ultimately optional if you wish to use Xata only as a Postgres host. If you do decide to utilize them (which is the default way Xata operates through our UI), it is required that you interact with these types through the Xata SDKs.
```
Comment (philkra) [inline 030-Data-types/030-data-model.mdx:10]: Maybe `text` is not a good example, as it's ambiguous. In Xata we have the limitation of 1000 chars and in PG there is "none". I added a bit about this limitation in this https://github.com/xataio/mdx-docs/pull/281/commits/9565399fa9a5640479039042baa5a797ff4c3a04

File: 030-Data-types/030-data-model.mdx [hunk 2]
```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
...
+
+## Xata specific data types
+
+Below is a mapping of the additional data types Xata provides alongside Postgres types.
+
+| Xata data type          | PostgreSQL data type | Notes                              |
+| ----------------------- | -------------------- | ---------------------------------- |
+| [string](#string)       | text                 |                                    |
+| [text](#text)           | text                 | With constraints on maximum length |
```
Comment (philkra) [inline 030-Data-types/030-data-model.mdx:19]:
```suggestion
| [text](#text)           | text                 | With constraints on maximum length of 1000 characters |
```

File: 030-Data-types/030-data-model.mdx [hunk 3]
```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
...
+| [int](#int)             | bigint               |                                    |
+| [float](#float)         | double precision     |                                    |
+| [datetime](#datetime)   | timestamptz          |                                    |
+| [bool](#bool)           | boolean              |                                    |
+| [email](#email)         | text                 | Validated by Xata                  |
+| [multiple](#multiple)   | text[]               | Validated by Xata                  |
+| [file](#file)           | jsonb                | Validated by Xata                  |
```
Comment (philkra) [inline 030-Data-types/030-data-model.mdx:26]: @sorintm I know the base type for `file` and `filearray` is `jsonb` but is that the type we show when you describe a table? I remember we have a custom type, à la `xata_file`. Can you confirm what is visible to the user for `file` and `filearray`? Thank you

File: 030-Data-types/030-data-model.mdx [hunk 4]
```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
...
+| [filearray](#filearray) | jsonb                | Validated by Xata                  |
+| [link](#link)           | foreign key          |                                    |
+| [vector](#vector)       | real[]               |                                    |
+
+## Limitations when using native Postgres types
+
+As you are able to create your own types or use types from extensions through the [Postgres wire protocol](/docs/postgres), there is a possibility that the Xata UI is not aware of the type. For such cases, the UI will display the column values as a string and the corresponding edit UI set to read-only. For now, manipulation of these columns needs to be performed through the wire protocol. Reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.
```
Comment (philkra) [inline 030-Data-types/030-data-model.mdx:33]:
```suggestion
As you are able to create your own types or use types from extensions in a [dedicated cluster](/docs/dedicated-cluster) through the [Postgres wire protocol](/docs/postgres), there is a possibility that the Xata UI is not aware of the type. For such cases, the UI will display the column values as a string and the corresponding edit UI set to read-only. For now, manipulation of these columns needs to be performed through the wire protocol. Reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.
```

Comment (philkra) [reply 030-Data-types/030-data-model.mdx:33]: These operations are only possible in a dedicated cluster.

File: 030-Data-types/030-data-model.mdx [hunk 5]
```diff
@@ -7,6 +7,33 @@ slug: concepts/data-model
...
+| [multiple](#multiple)   | text[]               | Validated by Xata                                     |
+| [file](#file)           | jsonb                | Validated by Xata                                     |
+| [filearray](#filearray) | jsonb                | Validated by Xata                                     |
+| [link](#link)           | foreign key          |                                                       |
+| [vector](#vector)       | real[]               |                                                       |
+
+## Limitations when using native Postgres types
+
+As you are able to create your own types or use types from extensions in a [dedicated cluster](/docs/dedicated-cluster) through the [Postgres wire protocol](/docs/postgres), there is a possibility that the Xata UI is not aware of the type. For such cases, the UI will display the column values as a string and the corresponding edit UI set to read-only. For now, manipulation of these columns needs to be performed through the wire protocol. Reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.
```
Comment (tsg) [inline 030-Data-types/030-data-model.mdx:33]: For:

> Reach out to our [support team](/docs/resources/support) to request adding the data type to Xata.

This sounded a bit unclear. Perhaps: ... to request adding support for a particular data type to Xata?

**Review by tsg — APPROVED** [review]

File: 030-Data-types/030-data-model.mdx (hunk 5 above)
Comment (snide) [reply 030-Data-types/030-data-model.mdx:33]: I think we should just remove this line completely. I'm not sure support will be able to do anything.

File: 030-Data-types/030-data-model.mdx (hunk 1 above)
Comment (snide) [reply 030-Data-types/030-data-model.mdx]: Added a link to this issue when discussing limitations in the section below. Also cleaned up the language.

File: 030-Data-types/030-data-model.mdx (hunk 3 above)
Comment (snide) [reply 030-Data-types/030-data-model.mdx]: Gonna merge this for now. I'll edit later when @sorintm adds context.

**Pull request merged**